In [1]:
import numpy as np 
import pandas as pd 
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
from matplotlib.pyplot import figure
from matplotlib.pyplot import suptitle
import matplotlib.style as style
from IPython.display import display, HTML
import warnings
%config InlineBackend.figure_format = 'png' #set 'png' here when working on notebook
warnings.filterwarnings('ignore') 

# Set some parameters to get good visuals - style to ggplot and size to 15,10

pd.set_option('display.width',170, 'display.max_rows',200, 'display.max_columns',900)

In [2]:
df = pd.read_csv("TADPOLE_D1_D2.csv")

In [3]:
print ("\n\n---------------------")
print ("Database INFORMATION")
print ("---------------------")
print ("Shape of training set:", df.shape, "\n")
print ("Column Headers:", list(df.columns.values), "\n")
print (df.dtypes)



---------------------
Database INFORMATION
---------------------
Shape of training set: (12741, 1907) 

Column Headers: ['RID', 'PTID', 'VISCODE', 'SITE', 'D1', 'D2', 'COLPROT', 'ORIGPROT', 'EXAMDATE', 'DX_bl', 'DXCHANGE', 'AGE', 'PTGENDER', 'PTEDUCAT', 'PTETHCAT', 'PTRACCAT', 'PTMARRY', 'APOE4', 'FDG', 'PIB', 'AV45', 'CDRSB', 'ADAS11', 'ADAS13', 'MMSE', 'RAVLT_immediate', 'RAVLT_learning', 'RAVLT_forgetting', 'RAVLT_perc_forgetting', 'FAQ', 'MOCA', 'EcogPtMem', 'EcogPtLang', 'EcogPtVisspat', 'EcogPtPlan', 'EcogPtOrgan', 'EcogPtDivatt', 'EcogPtTotal', 'EcogSPMem', 'EcogSPLang', 'EcogSPVisspat', 'EcogSPPlan', 'EcogSPOrgan', 'EcogSPDivatt', 'EcogSPTotal', 'FLDSTRENG', 'FSVERSION', 'Ventricles', 'Hippocampus', 'WholeBrain', 'Entorhinal', 'Fusiform', 'MidTemp', 'ICV', 'DX', 'EXAMDATE_bl', 'CDRSB_bl', 'ADAS11_bl', 'ADAS13_bl', 'MMSE_bl', 'RAVLT_immediate_bl', 'RAVLT_learning_bl', 'RAVLT_forgetting_bl', 'RAVLT_perc_forgetting_bl', 'FAQ_bl', 'FLDSTRENG_bl', 'FSVERSION_bl', 'Ventricles_bl', 

In [4]:
import re
missing_values = []
nonumeric_values = []

print ("Data INFORMATION")
print ("========================\n")

for column in df:
    # Find all the unique feature values
    uniq = df[column].unique()
    print ("'{}' has {} unique values" .format(column,uniq.size))
    if (uniq.size > 10):
        print("~~Listing up to 10 unique values~~")
    print (uniq[0:10])
    print ("\n-----------------------------------------------------------------------\n")
    
    # Find features with missing values
    if (True in pd.isnull(uniq)):
        s = "{} has {} missing" .format(column, pd.isnull(df[column]).sum())
        missing_values.append(s)
    
    # Find features with non-numeric values
    for i in range (1, np.prod(uniq.shape)):
        if (re.match('nan', str(uniq[i]))):
            break
        if not (re.search('(^\d+\.?\d*$)|(^\d*\.?\d+$)', str(uniq[i]))):
            nonumeric_values.append(column)
            break
  
print ("\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n")
print ("Features with missing values:\n{}\n\n" .format(missing_values))
print ("Features with non-numeric values:\n{}" .format(nonumeric_values))
print ("\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n")

Data INFORMATION

'RID' has 1737 unique values
~~Listing up to 10 unique values~~
[ 2  3  4  5  6  7  8 10 14 15]

-----------------------------------------------------------------------

'PTID' has 1737 unique values
~~Listing up to 10 unique values~~
['011_S_0002' '011_S_0003' '022_S_0004' '011_S_0005' '100_S_0006'
 '022_S_0007' '011_S_0008' '011_S_0010' '022_S_0014' '100_S_0015']

-----------------------------------------------------------------------

'VISCODE' has 22 unique values
~~Listing up to 10 unique values~~
['bl' 'm06' 'm12' 'm24' 'm18' 'm36' 'm72' 'm84' 'm48' 'm60']

-----------------------------------------------------------------------

'SITE' has 62 unique values
~~Listing up to 10 unique values~~
[ 11  22 100  67  23  35  99   7  18 123]

-----------------------------------------------------------------------

'D1' has 2 unique values
[1 0]

-----------------------------------------------------------------------

'D2' has 2 unique values
[1 0]

-----------------------

'EcogPtOrgan_bl' has 35 unique values
~~Listing up to 10 unique values~~
[    nan 1.16667 1.4     2.      1.5     2.16667 1.      3.8     1.83333
 1.33333]

-----------------------------------------------------------------------

'EcogPtDivatt_bl' has 16 unique values
~~Listing up to 10 unique values~~
[ nan 1.75 1.   2.25 1.25 1.5  3.25 2.   4.   2.5 ]

-----------------------------------------------------------------------

'EcogPtTotal_bl' has 224 unique values
~~Listing up to 10 unique values~~
[    nan 1.58974 1.60526 2.10256 1.34211 1.5641  1.17949 1.35897 2.86111
 1.66667]

-----------------------------------------------------------------------

'EcogSPMem_bl' has 48 unique values
~~Listing up to 10 unique values~~
[    nan 1.5     1.375   1.875   1.25    2.      3.      2.28571 3.25
 1.125  ]

-----------------------------------------------------------------------

'EcogSPLang_bl' has 60 unique values
~~Listing up to 10 unique values~~
[    nan 1.66667 1.44444 1.11111 1.55556 1

'ST108TA_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 810 unique values
~~Listing up to 10 unique values~~
[' ' '1.556' '1.582' '1.893' '1.882' '1.773' '1.828' '1.822' '1.851'
 '1.847']

-----------------------------------------------------------------------

'ST108TS_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 382 unique values
~~Listing up to 10 unique values~~
[' ' '0.691' '0.638' '0.704' '0.656' '0.681' '0.664' '0.657' '0.65'
 '0.613']

-----------------------------------------------------------------------

'ST109CV_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 1702 unique values
~~Listing up to 10 unique values~~
[' ' '4034' '3741' '3562' '3506' '3409' '3601' '3566' '3791' '3825']

-----------------------------------------------------------------------

'ST109SA_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 809 unique values
~~Listing up to 10 unique values~~
[' ' '1560' '1818' '1409' '1428' '1460' '1427' '1363' '1484' '1469']

---------------------------------------------------------

'ST121TS_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 427 unique values
~~Listing up to 10 unique values~~
[' ' '0.57' '0.633' '0.41' '0.416' '0.484' '0.435' '0.392' '0.431' '0.492']

-----------------------------------------------------------------------

'ST122SV_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 1 unique values
[' ']

-----------------------------------------------------------------------

'ST123CV_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 1789 unique values
~~Listing up to 10 unique values~~
[' ' '6304' '6327' '6316' '6378' '6599' '6414' '6285' '6775' '6788']

-----------------------------------------------------------------------

'ST123SA_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 1496 unique values
~~Listing up to 10 unique values~~
[' ' '7047' '7259' '5481' '5539' '5520' '5519' '5528' '5543' '5540']

-----------------------------------------------------------------------

'ST123TA_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 651 unique values
~~Listing up to 10 uniqu

'ST24TS_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 676 unique values
~~Listing up to 10 unique values~~
[' ' '1.025' '0.88' '0.956' '1.056' '0.911' '0.885' '0.897' '0.871'
 '1.003']

-----------------------------------------------------------------------

'ST25CV_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 726 unique values
~~Listing up to 10 unique values~~
[' ' '545' '526' '521' '466' '414' '427' '420' '755' '690']

-----------------------------------------------------------------------

'ST25SA_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 248 unique values
~~Listing up to 10 unique values~~
[' ' '157' '172' '139' '127' '130' '118' '116' '216' '206']

-----------------------------------------------------------------------

'ST25TA_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 1284 unique values
~~Listing up to 10 unique values~~
[' ' '2.526' '2.397' '2.826' '2.623' '2.486' '2.652' '2.635' '2.78'
 '2.695']

-----------------------------------------------------------------------

'ST25TS

'ST40TA_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 1090 unique values
~~Listing up to 10 unique values~~
[' ' '2.514' '2.447' '2.581' '2.589' '2.622' '2.552' '2.527' '2.847'
 '2.84']

-----------------------------------------------------------------------

'ST40TS_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 396 unique values
~~Listing up to 10 unique values~~
[' ' '0.655' '0.635' '0.623' '0.587' '0.599' '0.598' '0.577' '0.716'
 '0.76']

-----------------------------------------------------------------------

'ST41SV_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 1 unique values
[' ']

-----------------------------------------------------------------------

'ST42SV_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 1133 unique values
~~Listing up to 10 unique values~~
[' ' '1445' '1306' '1586' '1525' '1524' '1468' '1478' '1768' '1805']

-----------------------------------------------------------------------

'ST43CV_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 1781 unique values
~~Listing up to 1

'ST56SA_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 2386 unique values
~~Listing up to 10 unique values~~
[' ' '7650' '7983' '7880' '7798' '7645' '7708' '7780' '6992' '6898']

-----------------------------------------------------------------------

'ST56TA_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 918 unique values
~~Listing up to 10 unique values~~
[' ' '2.447' '2.303' '2.491' '2.403' '2.281' '2.379' '2.391' '2.481'
 '2.505']

-----------------------------------------------------------------------

'ST56TS_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 322 unique values
~~Listing up to 10 unique values~~
[' ' '0.661' '0.614' '0.667' '0.618' '0.617' '0.602' '0.664' '0.626'
 '0.625']

-----------------------------------------------------------------------

'ST57CV_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 3218 unique values
~~Listing up to 10 unique values~~
[' ' '11381' '11216' '11324' '11332' '10297' '10776' '11235' '11030'
 '11057']

-------------------------------------------------

'ST79SV_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 2495 unique values
~~Listing up to 10 unique values~~
[' ' '321681' '311876' '291545' '288656' '283116' '279142' '275133'
 '311743' '318749']

-----------------------------------------------------------------------

'ST7SV_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 1408 unique values
~~Listing up to 10 unique values~~
[' ' '1891' '2013' '1395' '1481' '1456' '1385' '1434' '1430' '1424']

-----------------------------------------------------------------------

'ST80SV_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 1733 unique values
~~Listing up to 10 unique values~~
[' ' '3748' '3945' '3002' '3106' '2979' '3035' '3033' '2265' '2097']

-----------------------------------------------------------------------

'ST81CV_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 1 unique values
[' ']

-----------------------------------------------------------------------

'ST81SA_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 1 unique values
[' ']

------------

'ST98CV_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 2168 unique values
~~Listing up to 10 unique values~~
[' ' '4046' '4158' '3925' '3838' '3777' '3514' '3701' '4551' '4515']

-----------------------------------------------------------------------

'ST98SA_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 1044 unique values
~~Listing up to 10 unique values~~
[' ' '1947' '2171' '1683' '1638' '1652' '1676' '1669' '1723' '1712']

-----------------------------------------------------------------------

'ST98TA_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 938 unique values
~~Listing up to 10 unique values~~
[' ' '2.021' '1.991' '2.159' '2.138' '2.106' '1.994' '2.071' '2.459'
 '2.473']

-----------------------------------------------------------------------

'ST98TS_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16' has 523 unique values
~~Listing up to 10 unique values~~
[' ' '0.745' '0.678' '0.627' '0.656' '0.666' '0.598' '0.736' '0.752'
 '0.739']

-----------------------------------------------------------

'ST108SA_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 3298 unique values
~~Listing up to 10 unique values~~
[4221 3679 3712 3718 3703 4248 4251 4323 4220 4344]

-----------------------------------------------------------------------

'ST108TA_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 1576 unique values
~~Listing up to 10 unique values~~
[1.766 1.675 1.641 1.6380000000000001 1.646 1.9340000000000002
 1.8869999999999998 1.7730000000000001 1.8519999999999999 1.787]

-----------------------------------------------------------------------

'ST108TS_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 751 unique values
~~Listing up to 10 unique values~~
[0.723 0.685 0.631 0.648 0.68 0.682 0.645 0.6609999999999999 0.659 0.6]

-----------------------------------------------------------------------

'ST109CV_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 3159 unique values
~~Listing up to 10 unique values~~
[3809 4003 4106 4004 3852 3559 3449 3339 3364 3910]

----------------------------------------------------------

'ST118TA_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 1840 unique values
~~Listing up to 10 unique values~~
[2.279 2.289 2.259 2.2319999999999998 2.2119999999999997 2.226 2.27 2.164
 2.07 2.116]

-----------------------------------------------------------------------

'ST118TS_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 678 unique values
~~Listing up to 10 unique values~~
[0.787 0.657 0.669 0.65 0.67 0.555 0.598 0.62 0.578 0.541]

-----------------------------------------------------------------------

'ST119CV_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 2970 unique values
~~Listing up to 10 unique values~~
[1765 1341 1159 1350 1279 1442 1548 1620 1525 1477]

-----------------------------------------------------------------------

'ST119SA_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 733 unique values
~~Listing up to 10 unique values~~
[297 309 368 329 325 270 292 313 330 323]

-----------------------------------------------------------------------

'ST119TA_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 

'ST19SV_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 3856 unique values
~~Listing up to 10 unique values~~
['250839' '191321' '187866' '184005' '184689' '222637' '210846' '211047'
 '203957' '204616']

-----------------------------------------------------------------------

'ST1SV_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 6318 unique values
~~Listing up to 10 unique values~~
[23003 24577 23784 23759 23324 22145 21109 21327 20843 20232]

-----------------------------------------------------------------------

'ST20SV_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 3863 unique values
~~Listing up to 10 unique values~~
['259916' '249429' '240068' '243148' '233306' '254050' '250900' '251687'
 '249057' '246864']

-----------------------------------------------------------------------

'ST21SV_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 3026 unique values
~~Listing up to 10 unique values~~
[2512 3544 3427 3210 3777 2640 2772 2887 2703 3018]

------------------------------------------------------------------

'ST37SV_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 7514 unique values
~~Listing up to 10 unique values~~
[57742 43635 45561 46081 49553 19569 19507 18706 19691 21851]

-----------------------------------------------------------------------

'ST38CV_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 4822 unique values
~~Listing up to 10 unique values~~
[6644 5542 5232 5279 5402 7571 7000 6914 6898 6819]

-----------------------------------------------------------------------

'ST38SA_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 3077 unique values
~~Listing up to 10 unique values~~
[3103 3125 2958 3049 3036 3994 3858 3873 3860 3920]

-----------------------------------------------------------------------

'ST38TA_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 1383 unique values
~~Listing up to 10 unique values~~
[1.931 1.625 1.6269999999999998 1.59 1.6280000000000001 1.804 1.724 1.71
 1.723 1.662]

-----------------------------------------------------------------------

'ST38TS_UCSFFSX_11_02_15_UCSFFSX51_08_0

'ST49TS_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 776 unique values
~~Listing up to 10 unique values~~
[0.691 0.6940000000000001 0.688 0.6729999999999999 0.684 0.56 0.546 0.522
 0.535 0.532]

-----------------------------------------------------------------------

'ST4SV_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 815 unique values
~~Listing up to 10 unique values~~
[432 341 358 323 317 475 446 457 454 422]

-----------------------------------------------------------------------

'ST50CV_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 3233 unique values
~~Listing up to 10 unique values~~
[3614 2612 2534 2623 2609 3270 3327 3169 3422 3046]

-----------------------------------------------------------------------

'ST50SA_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 1492 unique values
~~Listing up to 10 unique values~~
[1267 1082 1080 1066 1044 1350 1423 1361 1456 1358]

-----------------------------------------------------------------------

'ST50TA_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 1808 unique 

'ST59TS_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 684 unique values
~~Listing up to 10 unique values~~
[0.551 0.563 0.573 0.624 0.598 0.578 0.5870000000000001 0.637 0.523 0.545]

-----------------------------------------------------------------------

'ST5SV_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 789 unique values
~~Listing up to 10 unique values~~
[404 307 314 319 385 381 372 354 360 368]

-----------------------------------------------------------------------

'ST60CV_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 3138 unique values
~~Listing up to 10 unique values~~
[1635 1813 1547 1536 1521 1388 1301 1295 1327 1564]

-----------------------------------------------------------------------

'ST60SA_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 759 unique values
~~Listing up to 10 unique values~~
[355 325 326 370 379 322 330 333 372 323]

-----------------------------------------------------------------------

'ST60TA_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 3012 unique values
~~Listing up to 1

'ST83TS_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 1423 unique values
~~Listing up to 10 unique values~~
[0.925 0.915 0.909 0.887 0.8909999999999999 1.072 0.9059999999999999 1.03
 0.8490000000000001 1.056]

-----------------------------------------------------------------------

'ST84CV_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 1789 unique values
~~Listing up to 10 unique values~~
[725 740 798 714 816 685 728 535 730 687]

-----------------------------------------------------------------------

'ST84SA_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 565 unique values
~~Listing up to 10 unique values~~
[244 236 269 249 245 252 247 239 260 173]

-----------------------------------------------------------------------

'ST84TA_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 2411 unique values
~~Listing up to 10 unique values~~
[2.388 2.603 2.426 2.359 2.64 2.286 2.343 1.82 2.4130000000000003
 2.2840000000000003]

-----------------------------------------------------------------------

'ST84TS_UCSFFSX_11_02

'ST99SA_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 3142 unique values
~~Listing up to 10 unique values~~
[4272 3316 3138 3240 3097 3598 3804 3757 3630 3733]

-----------------------------------------------------------------------

'ST99TA_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 2056 unique values
~~Listing up to 10 unique values~~
[3.028 2.545 2.445 2.502 2.474 2.67 2.654 2.5610000000000004 2.555
 2.5660000000000003]

-----------------------------------------------------------------------

'ST99TS_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 802 unique values
~~Listing up to 10 unique values~~
[0.649 0.6859999999999999 0.713 0.685 0.677 0.631 0.65 0.619 0.68 0.66]

-----------------------------------------------------------------------

'ST9SV_UCSFFSX_11_02_15_UCSFFSX51_08_01_16' has 3953 unique values
~~Listing up to 10 unique values~~
[4396 3304 3173 3235 3314 1338 1365 1016 1422 1268]

-----------------------------------------------------------------------

'update_stamp_UCSFFSX_11_02_15

'TMPMIDR03_BAIPETNMRC_09_12_16' has 95 unique values
~~Listing up to 10 unique values~~
['-4' '1.3553' '1.32432' ' ' '1.18593' '1.15367' nan '1.28962' '1.31035'
 '1.16309']

-----------------------------------------------------------------------

'TMPINFL03_BAIPETNMRC_09_12_16' has 95 unique values
~~Listing up to 10 unique values~~
['-4' '1.12063' '1.05657' ' ' '1.21911' '1.19895' nan '1.18311' '1.12248'
 '1.25653']

-----------------------------------------------------------------------

'TMPINFR03_BAIPETNMRC_09_12_16' has 95 unique values
~~Listing up to 10 unique values~~
['-4' '1.29457' '1.25993' ' ' '1.19541' '1.14196' nan '1.22947' '1.24428'
 '1.10709']

-----------------------------------------------------------------------

'FUSFRML03_BAIPETNMRC_09_12_16' has 3 unique values
['-4' ' ' nan]

-----------------------------------------------------------------------

'FUSFRMR03_BAIPETNMRC_09_12_16' has 3 unique values
['-4' ' ' nan]

------------------------------------------------

'OCCINFR05_BAIPETNMRC_09_12_16' has 3 unique values
['-4' ' ' nan]

-----------------------------------------------------------------------

'THALAML05_BAIPETNMRC_09_12_16' has 3 unique values
['-4' ' ' nan]

-----------------------------------------------------------------------

'THALAMR05_BAIPETNMRC_09_12_16' has 3 unique values
['-4' ' ' nan]

-----------------------------------------------------------------------

'FRTMEDL05_BAIPETNMRC_09_12_16' has 3 unique values
['-4' ' ' nan]

-----------------------------------------------------------------------

'FRTMEDR05_BAIPETNMRC_09_12_16' has 3 unique values
['-4' ' ' nan]

-----------------------------------------------------------------------

'ANGULR05_BAIPETNMRC_09_12_16' has 313 unique values
~~Listing up to 10 unique values~~
['-4' ' ' nan '1.30146' '1.28959' '1.25527' '1.25266' '1.30581' '1.27691'
 '1.34249']

-----------------------------------------------------------------------

'ANGULL05_BAIPETNMRC_09_12_16' has 3 unique val

'NL1PTEMP_BAIPETNMRC_09_12_16' has 119 unique values
~~Listing up to 10 unique values~~
['-4' '-0.009185' '0.062646' ' ' '0.008374' '0.234965' nan '-0.03398'
 '0.142959' '0.24458']

-----------------------------------------------------------------------

'NL1PSTCG_BAIPETNMRC_09_12_16' has 119 unique values
~~Listing up to 10 unique values~~
['-4' '0.048075' '0.02179' ' ' '0.0978' '0.174251' nan '-0.017056'
 '0.211739' '-0.114747']

-----------------------------------------------------------------------

'NL2PSTCG_BAIPETNMRC_09_12_16' has 260 unique values
~~Listing up to 10 unique values~~
['-4' ' ' nan '-0.035535' '0.025473' '0.019717' '0.13808' '0.098566'
 '0.049137' '0.030109']

-----------------------------------------------------------------------

'NL2PARI_BAIPETNMRC_09_12_16' has 132 unique values
~~Listing up to 10 unique values~~
['-4' ' ' nan '0.071205' '0.004865' '0.122659' '0.077672' '0.019305'
 '-0.002499' '0.075027']

------------------------------------------------------

'CC_MID_POSTERIOR_SIZE_UCBERKELEYAV45_10_17_16' has 328 unique values
~~Listing up to 10 unique values~~
[' ' '272' '290' '300' '223' '424' '264' '547' '303' '258']

-----------------------------------------------------------------------

'CC_POSTERIOR_UCBERKELEYAV45_10_17_16' has 1937 unique values
~~Listing up to 10 unique values~~
[' ' '2.0514' '1.6886' '2.1411' '2.4519' '2.1117' '1.7725' '2.0675'
 '1.6737' '1.2911']

-----------------------------------------------------------------------

'CC_POSTERIOR_SIZE_UCBERKELEYAV45_10_17_16' has 506 unique values
~~Listing up to 10 unique values~~
[' ' '768' '805' '839' '822' '975' '733' '1103' '778' '665']

-----------------------------------------------------------------------

'CSF_UCBERKELEYAV45_10_17_16' has 1763 unique values
~~Listing up to 10 unique values~~
[' ' '0.7148' '0.5589' '0.8472' '1.0921' '0.9113' '0.7872' '0.9244'
 '0.8068' '0.6002']

-----------------------------------------------------------------------

'CSF_SIZE_UCBERK

'CTX_LH_POSTERIORCINGULATE_SIZE_UCBERKELEYAV45_10_17_16' has 794 unique values
~~Listing up to 10 unique values~~
[' ' '2732' '2306' '2525' '2336' '2484' '2641' '2980' '2113' '2286']

-----------------------------------------------------------------------

'CTX_LH_PRECENTRAL_UCBERKELEYAV45_10_17_16' has 1823 unique values
~~Listing up to 10 unique values~~
[' ' '1.2204' '1.2122' '1.2207' '1.3378' '1.3491' '1.255' '1.1788'
 '1.1404' '0.984']

-----------------------------------------------------------------------

'CTX_LH_PRECENTRAL_SIZE_UCBERKELEYAV45_10_17_16' has 994 unique values
~~Listing up to 10 unique values~~
[' ' '11459' '8531' '8571' '8005' '10904' '8418' '12170' '9933' '10836']

-----------------------------------------------------------------------

'CTX_LH_PRECUNEUS_UCBERKELEYAV45_10_17_16' has 1919 unique values
~~Listing up to 10 unique values~~
[' ' '1.541' '1.2176' '1.2764' '1.3373' '1.3379' '1.5949' '1.2416'
 '1.1218' '1.0282']

---------------------------------------

'CTX_RH_MEDIALORBITOFRONTAL_UCBERKELEYAV45_10_17_16' has 1906 unique values
~~Listing up to 10 unique values~~
[' ' '1.2034' '1.006' '1.1103' '1.2065' '1.2463' '1.4715' '1.1469'
 '1.1296' '0.828']

-----------------------------------------------------------------------

'CTX_RH_MEDIALORBITOFRONTAL_SIZE_UCBERKELEYAV45_10_17_16' has 836 unique values
~~Listing up to 10 unique values~~
[' ' '4145' '3647' '4011' '3416' '4775' '3680' '4425' '3965' '4692']

-----------------------------------------------------------------------

'CTX_RH_MIDDLETEMPORAL_UCBERKELEYAV45_10_17_16' has 1891 unique values
~~Listing up to 10 unique values~~
[' ' '1.3444' '1.0527' '1.1518' '1.345' '1.2194' '1.454' '1.0389' '1.0671'
 '0.9781']

-----------------------------------------------------------------------

'CTX_RH_MIDDLETEMPORAL_SIZE_UCBERKELEYAV45_10_17_16' has 998 unique values
~~Listing up to 10 unique values~~
[' ' '10561' '7714' '7469' '7089' '11265' '8405' '12074' '10923' '10475']

--------------------

'LEFT_CEREBELLUM_WHITE_MATTER_SIZE_UCBERKELEYAV45_10_17_16' has 1009 unique values
~~Listing up to 10 unique values~~
[' ' '14622' '10290' '14392' '12780' '13521' '9689' '12900' '11129'
 '12711']

-----------------------------------------------------------------------

'LEFT_CEREBRAL_WHITE_MATTER_UCBERKELEYAV45_10_17_16' has 1872 unique values
~~Listing up to 10 unique values~~
[' ' '1.8528' '1.6321' '1.8847' '2.0427' '1.9601' '1.8259' '1.7751'
 '1.6105' '1.3636']

-----------------------------------------------------------------------

'LEFT_CEREBRAL_WHITE_MATTER_SIZE_UCBERKELEYAV45_10_17_16' has 1091 unique values
~~Listing up to 10 unique values~~
[' ' '209164' '192694' '194885' '174755' '263831' '219865' '210901'
 '234019' '215958']

-----------------------------------------------------------------------

'LEFT_CHOROID_PLEXUS_UCBERKELEYAV45_10_17_16' has 1832 unique values
~~Listing up to 10 unique values~~
[' ' '1.029' '0.9423' '1.2508' '1.5291' '1.0695' '0.9626' '1.1341'
 '0.9483

'RIGHT_VENTRALDC_SIZE_UCBERKELEYAV45_10_17_16' has 779 unique values
~~Listing up to 10 unique values~~
[' ' '3726' '2574' '3246' '2911' '3791' '3476' '3554' '3633' '4040']

-----------------------------------------------------------------------

'RIGHT_VESSEL_UCBERKELEYAV45_10_17_16' has 1857 unique values
~~Listing up to 10 unique values~~
[' ' '1.7882' '1.3074' '1.4963' '1.3669' '1.6753' '1.4432' '1.4312'
 '1.5398' '1.1649']

-----------------------------------------------------------------------

'RIGHT_VESSEL_SIZE_UCBERKELEYAV45_10_17_16' has 207 unique values
~~Listing up to 10 unique values~~
[' ' '46' '165' '105' '88' '103' '80' '63' '65' '57']

-----------------------------------------------------------------------

'WM_HYPOINTENSITIES_UCBERKELEYAV45_10_17_16' has 1887 unique values
~~Listing up to 10 unique values~~
[' ' '1.8298' '1.6873' '1.8781' '2.2011' '1.7591' '1.4855' '1.7104'
 '1.5055' '1.2649']

-----------------------------------------------------------------------



'CTX_RH_POSTCENTRAL_UCBERKELEYAV1451_10_17_16' has 88 unique values
~~Listing up to 10 unique values~~
[' ' '0.8757' '0.8508' '0.9499' '1.0984' '0.8124' '0.8394' '0.8179'
 '0.8649' '0.9778']

-----------------------------------------------------------------------

'CTX_RH_POSTCENTRAL_SIZE_UCBERKELEYAV1451_10_17_16' has 85 unique values
~~Listing up to 10 unique values~~
[' ' '6688' '6660' '9278' '8045' '5802' '9387' '8058' '8074' '9137']

-----------------------------------------------------------------------

'CTX_RH_POSTERIORCINGULATE_UCBERKELEYAV1451_10_17_16' has 90 unique values
~~Listing up to 10 unique values~~
[' ' '1.0335' '1.0087' '1.139' '1.3716' '0.9481' '1.0025' '1.0033'
 '0.9672' '1.0905']

-----------------------------------------------------------------------

'CTX_RH_POSTERIORCINGULATE_SIZE_UCBERKELEYAV1451_10_17_16' has 85 unique values
~~Listing up to 10 unique values~~
[' ' '2474' '2756' '4018' '2570' '2967' '2322' '2699' '3118' '2103']

----------------------------

'FA_FX_ST_R_DTIROI_04_30_14' has 787 unique values
~~Listing up to 10 unique values~~
[' ' '0.26174' '0.2268' '0.315287' '0.317036' '0.299998' '0.29342'
 '0.280287' '0.285995' '0.348533']

-----------------------------------------------------------------------

'FA_SLF_L_DTIROI_04_30_14' has 784 unique values
~~Listing up to 10 unique values~~
[' ' '0.323927' '0.338162' '0.338817' '0.33894' '0.345403' '0.324109'
 '0.336302' '0.322837' '0.352024']

-----------------------------------------------------------------------

'FA_SLF_R_DTIROI_04_30_14' has 788 unique values
~~Listing up to 10 unique values~~
[' ' '0.347582' '0.341044' '0.368017' '0.356808' '0.376689' '0.362915'
 '0.36704' '0.350958' '0.365288']

-----------------------------------------------------------------------

'FA_SFO_L_DTIROI_04_30_14' has 788 unique values
~~Listing up to 10 unique values~~
[' ' '0.208568' '0.217421' '0.210334' '0.228148' '0.214534' '0.200088'
 '0.183904' '0.258521' '0.253554']

---------------------

'MD_BCC_R_DTIROI_04_30_14' has 786 unique values
~~Listing up to 10 unique values~~
[' ' '0.00162083' '0.00165547' '0.00119042' '0.00120146' '0.00128163'
 '0.00123685' '0.00128352' '0.00116217' '0.00113409']

-----------------------------------------------------------------------

'MD_SCC_L_DTIROI_04_30_14' has 780 unique values
~~Listing up to 10 unique values~~
[' ' '0.00127693' '0.00127712' '0.00102705' '0.001002' '0.00108503'
 '0.00108345' '0.00107642' '0.00104033' '0.00095146']

-----------------------------------------------------------------------

'MD_SCC_R_DTIROI_04_30_14' has 786 unique values
~~Listing up to 10 unique values~~
[' ' '0.00114603' '0.00113142' '0.00100459' '0.00099244' '0.00102834'
 '0.0010196' '0.00106348' '0.00094714' '0.00092642']

-----------------------------------------------------------------------

'MD_RLIC_L_DTIROI_04_30_14' has 775 unique values
~~Listing up to 10 unique values~~
[' ' '0.00086552' '0.000851356' '0.00084234' '0.00083698' '0.00084104'
 

'AD_ML_L_DTIROI_04_30_14' has 777 unique values
~~Listing up to 10 unique values~~
[' ' '0.00126938' '0.00133053' '0.00123696' '0.001313' '0.00120549'
 '0.00120195' '0.00123498' '0.00120083' '0.00114502']

-----------------------------------------------------------------------

'AD_ML_R_DTIROI_04_30_14' has 778 unique values
~~Listing up to 10 unique values~~
[' ' '0.00128226' '0.00134379' '0.00130036' '0.00132413' '0.00129976'
 '0.00130194' '0.00131329' '0.00123754' '0.00109322']

-----------------------------------------------------------------------

'AD_SCP_L_DTIROI_04_30_14' has 783 unique values
~~Listing up to 10 unique values~~
[' ' '0.00148106' '0.00174992' '0.00156069' '0.00160257' '0.00157641'
 '0.00158225' '0.001692' '0.00141005' '0.00148538']

-----------------------------------------------------------------------

'AD_SCP_R_DTIROI_04_30_14' has 784 unique values
~~Listing up to 10 unique values~~
[' ' '0.00157269' '0.00151939' '0.00164844' '0.00168051' '0.00156315'
 '0.00

In [126]:
df.head(20)

RID        PTID VISCODE  SITE  D1  D2 COLPROT ORIGPROT    EXAMDATE DX_bl  DXCHANGE   AGE PTGENDER  PTEDUCAT         PTETHCAT PTRACCAT  PTMARRY  APOE4      FDG  PIB  \
0     2  011_S_0002      bl    11   1   1   ADNI1    ADNI1  2005-09-08    CN       1.0  74.3     Male        16  Not Hisp/Latino    White  Married    0.0  1.36926  NaN   
1     3  011_S_0003      bl    11   1   0   ADNI1    ADNI1  2005-09-12    AD       3.0  81.3     Male        18  Not Hisp/Latino    White  Married    1.0  1.09079  NaN   
2     3  011_S_0003     m06    11   1   0   ADNI1    ADNI1  2006-03-13    AD       3.0  81.3     Male        18  Not Hisp/Latino    White  Married    1.0  1.06360  NaN   
3     3  011_S_0003     m12    11   1   0   ADNI1    ADNI1  2006-09-12    AD       3.0  81.3     Male        18  Not Hisp/Latino    White  Married    1.0  1.10384  NaN   
4     3  011_S_0003     m24    11   1   0   ADNI1    ADNI1  2007-09-12    AD       3.0  81.3     Male        18  Not Hisp/Latino    White  Married    1.0  1.03871  NaN   
5     4  022_S_0004      bl    22   1   0   ADNI1    ADNI1  2005-11-08  LMCI       2.0  67.5     Male        10      Hisp/Latino    White  Married    0.0      NaN  NaN   
6     4  022_S_0004     m06    22   1   0   ADNI1    ADNI1  2006-05-02  LMCI       2.0  67.5     Male        10      Hisp/Latino    White  Married    0.0      NaN  NaN   
7     4  022_S_0004     m12    22   1   0   ADNI1    ADNI1  2006-11-14  LMCI       2.0  67.5     Male        10      Hisp/Latino    White  Married    0.0      NaN  NaN   
8     4  022_S_0004     m18    22   1   0   ADNI1    ADNI1  2007-05-14  LMCI       2.0  67.5     Male        10      Hisp/Latino    White  Married    0.0      NaN  NaN   
9     4  022_S_0004     m36    22   1   0   ADNI1    ADNI1  2008-11-18  LMCI       2.0  67.5     Male        10      Hisp/Latino    White  Married    0.0      NaN  NaN   
10    5  011_S_0005      bl    11   1   0   ADNI1    ADNI1  2005-09-07    CN       1.0  73.7     Male        16  Not Hisp/Latino    White  Married    0.0  1.29799  NaN   
11    5  011_S_0005     m06    11   1   0   ADNI1    ADNI1  2006-03-09    CN       1.0  73.7     Male        16  Not Hisp/Latino    White  Married    0.0  1.30617  NaN   
12    5  011_S_0005     m12    11   1   0   ADNI1    ADNI1  2006-09-05    CN       1.0  73.7     Male        16  Not Hisp/Latino    White  Married    0.0  1.28474  NaN   
13    5  011_S_0005     m24    11   1   0   ADNI1    ADNI1  2007-09-07    CN       1.0  73.7     Male        16  Not Hisp/Latino    White  Married    0.0  1.36742  NaN   
14    5  011_S_0005     m36    11   1   0   ADNI1    ADNI1  2008-09-10    CN       1.0  73.7     Male        16  Not Hisp/Latino    White  Married    0.0  1.34213  NaN   
15    6  100_S_0006      bl   100   1   0   ADNI1    ADNI1  2005-11-29  LMCI       2.0  80.4   Female        13  Not Hisp/Latino    White  Married    0.0      NaN  NaN   
16    6  100_S_0006     m06   100   1   0   ADNI1    ADNI1  2006-06-01  LMCI       2.0  80.4   Female        13  Not Hisp/Latino    White  Married    0.0      NaN  NaN   
17    6  100_S_0006     m12   100   1   0   ADNI1    ADNI1  2006-11-20  LMCI       2.0  80.4   Female        13  Not Hisp/Latino    White  Married    0.0      NaN  NaN   
18    6  100_S_0006     m18   100   1   0   ADNI1    ADNI1  2007-05-15  LMCI       2.0  80.4   Female        13  Not Hisp/Latino    White  Married    0.0      NaN  NaN   
19    6  100_S_0006     m36   100   1   0   ADNI1    ADNI1  2008-12-08  LMCI       2.0  80.4   Female        13  Not Hisp/Latino    White  Married    0.0      NaN  NaN   

    AV45  CDRSB  ADAS11  ADAS13  MMSE  RAVLT_immediate  RAVLT_learning  RAVLT_forgetting  RAVLT_perc_forgetting   FAQ  MOCA  EcogPtMem  EcogPtLang  EcogPtVisspat  \
0    NaN    0.0   10.67   18.67  28.0             44.0             4.0               6.0                54.5455   0.0   NaN        NaN         NaN            NaN   
1    NaN    4.5   22.00   31.00  20.0             22.0             1.0            

In [5]:
df['DX'].value_counts()

MCI                3932
NL                 2668
Dementia           1732
MCI to Dementia     372
NL to MCI           108
MCI to NL            77
Dementia to MCI      12
NL to Dementia        3
Name: DX, dtype: int64

In [6]:
df['PTGENDER'].value_counts()

Male      7217
Female    5524
Name: PTGENDER, dtype: int64

In [7]:
df.groupby('RID')['PTGENDER'].max().value_counts()

Male      957
Female    780
Name: PTGENDER, dtype: int64

In [8]:
df.groupby('RID')['AGE'].max(ascending=False).sort_values().value_counts().head(10)

75.8    16
74.3    15
70.8    15
76.1    15
71.4    15
80.4    14
70.9    14
73.5    14
76.4    14
69.3    13
Name: AGE, dtype: int64

In [9]:
df.groupby('RID')['AGE'].min().sort_values().value_counts().head(10)

75.8    16
74.3    15
70.8    15
76.1    15
71.4    15
80.4    14
70.9    14
73.5    14
76.4    14
69.3    13
Name: AGE, dtype: int64

#### Hence, There atr 957 males and 780 females

In [10]:
df.groupby('RID')['DX'].unique().head(20)

RID
2                      [NL, nan, NL to MCI]
3                           [Dementia, nan]
4                                [MCI, nan]
5                                 [NL, nan]
6                                [MCI, nan]
7                           [Dementia, nan]
8                      [NL, nan, NL to MCI]
10                          [Dementia, nan]
14                                [NL, nan]
15                     [NL, nan, NL to MCI]
16                                [NL, nan]
19                                [NL, nan]
21                                [NL, nan]
22                     [NL, NL to MCI, nan]
23                                [NL, nan]
29                          [Dementia, nan]
30    [MCI, MCI to Dementia, Dementia, nan]
31                                [NL, nan]
33                                    [MCI]
35                     [NL, nan, NL to MCI]
Name: DX, dtype: object

In [11]:
df[df['D1'] == 1].shape

(12671, 1907)

In [12]:
df[df['D1'] == 0].shape

(70, 1907)

In [13]:
df['RID'][df['D1'] == 0].unique()

array([ 360,  407,  436,  438,  445,  469,  506,  590,  592,  613,  640,
        653,  761,  777,  828,  871,  876, 1149, 1192, 1199, 1210, 1338,
       1357, 1435, 2003, 2199, 2278, 4066, 4085, 4186, 4260, 4362, 4524,
       4575, 4603, 4622, 4633, 4564, 4798, 4914, 4963, 5148, 5240, 5250,
       5262, 2351, 2011, 5206, 5279, 5146,  288,  339,  699,  884, 1184,
       1248, 1306, 1340, 1391, 4095, 4459, 4583, 4694, 4719, 4924, 4997,
       5120, 5241, 4264, 4740])

In [14]:
df['RID'][df['D1'] == 0].nunique()

70

#### Hence, 70 Patients did not roll from the study.

In [15]:
df[df['D2'] == 1].shape

(7660, 1907)

In [16]:
df.groupby('RID')['DX'].value_counts()

RID   DX             
2     NL                  7
      NL to MCI           1
3     Dementia            4
4     MCI                 5
5     NL                  5
6     MCI                 6
7     Dementia            4
8     NL                  9
      NL to MCI           1
10    Dementia            4
14    NL                  5
15    NL                  5
      NL to MCI           1
16    NL                  5
19    NL                  4
21    NL                 10
22    NL                  3
      NL to MCI           1
23    NL                 10
29    Dementia            4
30    Dementia            4
      MCI                 1
      MCI to Dementia     1
31    NL                 11
33    MCI                 6
35    NL                  5
      NL to MCI           1
38    MCI                 3
40    NL                  5
41    Dementia            3
      MCI                 3
      MCI to Dementia     1
42    Dementia            8
      MCI                 2
      MCI to Dementia     

### Chgecking if the diagnosis in both the training set and prediction test

In [17]:
df[['RID','DX']][df['D1'] == 1].groupby('RID')['DX'].value_counts()

RID   DX             
2     NL                  7
      NL to MCI           1
3     Dementia            4
4     MCI                 5
5     NL                  5
6     MCI                 6
7     Dementia            4
8     NL                  9
      NL to MCI           1
10    Dementia            4
14    NL                  5
15    NL                  5
      NL to MCI           1
16    NL                  5
19    NL                  4
21    NL                 10
22    NL                  3
      NL to MCI           1
23    NL                 10
29    Dementia            4
30    Dementia            4
      MCI                 1
      MCI to Dementia     1
31    NL                 11
33    MCI                 6
35    NL                  5
      NL to MCI           1
38    MCI                 3
40    NL                  5
41    Dementia            3
      MCI                 3
      MCI to Dementia     1
42    Dementia            8
      MCI                 2
      MCI to Dementia     

In [18]:
df[['PTID','DX']][df['D2'] == 1].groupby('PTID')['DX'].value_counts()

PTID        DX             
002_S_0295  NL                  8
002_S_0413  NL                 11
002_S_0685  NL                  8
002_S_1155  MCI                12
002_S_1261  NL                 10
002_S_1268  MCI                 8
            MCI to Dementia     1
002_S_1280  NL                 10
002_S_2043  MCI                 3
002_S_2073  MCI                 5
            NL                  2
            MCI to NL           1
002_S_4171  MCI                 3
            MCI to Dementia     1
002_S_4213  NL                  5
002_S_4219  MCI                 3
002_S_4225  NL                  5
002_S_4229  MCI                 7
002_S_4251  MCI                 2
002_S_4262  NL                  3
            MCI                 2
            Dementia            1
            NL to MCI           1
002_S_4270  NL                  4
            NL to MCI           1
002_S_4447  MCI                 5
002_S_4473  MCI                 6
002_S_4521  MCI                 4
            Dementia

In [19]:
df[['D1','D2','EXAMDATE','AGE','PTGENDER','PTEDUCAT','PTETHCAT','DX','EXAMDATE_bl','ADAS11_bl']][df['PTID'] =='002_S_2073']

D1  D2    EXAMDATE   AGE PTGENDER  PTEDUCAT         PTETHCAT         DX EXAMDATE_bl  ADAS11_bl
3122    1   1  2010-09-21  63.4   Female        20  Not Hisp/Latino        MCI  2010-09-21        6.0
3123    1   1  2010-12-06  63.4   Female        20  Not Hisp/Latino        NaN  2010-09-21        6.0
5524    1   1  2012-09-18  63.4   Female        20  Not Hisp/Latino        MCI  2010-09-21        6.0
5525    1   1  2011-09-29  63.4   Female        20  Not Hisp/Latino        MCI  2010-09-21        6.0
5526    1   1  2011-03-17  63.4   Female        20  Not Hisp/Latino        MCI  2010-09-21        6.0
5691    1   1  2013-10-01  63.4   Female        20  Not Hisp/Latino        MCI  2010-09-21        6.0
9145    1   1  2012-03-26  63.4   Female        20  Not Hisp/Latino        NaN  2010-09-21        6.0
9146    1   1  2013-04-02  63.4   Female        20  Not Hisp/Latino        NaN  2010-09-21        6.0
10934   1   1  2014-04-01  63.4   Female        20  Not Hisp/Latino        NaN  2010-09-21        6.0
11395   1   1  2014-10-07  63.4   Female        20  Not Hisp/Latino  MCI to NL  2010-09-21        6.0
12017   1   1  2015-09-29  63.4   Female        20  Not Hisp/Latino         NL  2010-09-21        6.0
12668   1   1  2016-09-27  63.4   Female        20  Not Hisp/Latino         NL  2010-09-21        6.0

In [20]:
df[['D1','D2','EXAMDATE','AGE','PTGENDER','PTEDUCAT','PTETHCAT','DX','EXAMDATE_bl','ADAS11_bl']][df['PTID'] =='002_S_0559']

D1  D2    EXAMDATE   AGE PTGENDER  PTEDUCAT         PTETHCAT   DX EXAMDATE_bl  ADAS11_bl
1328   1   0  2006-06-20  79.3     Male        16  Not Hisp/Latino   NL  2006-06-20        6.0
1329   1   0  2008-08-11  79.3     Male        16  Not Hisp/Latino   NL  2006-06-20        6.0
1330   1   0  2009-07-02  79.3     Male        16  Not Hisp/Latino   NL  2006-06-20        6.0
7130   1   0  2006-12-12  79.3     Male        16  Not Hisp/Latino   NL  2006-06-20        6.0
7131   1   0  2007-07-13  79.3     Male        16  Not Hisp/Latino   NL  2006-06-20        6.0
7132   1   0  2007-12-20  79.3     Male        16  Not Hisp/Latino  NaN  2006-06-20        6.0
7133   1   0  2008-12-09  79.3     Male        16  Not Hisp/Latino  NaN  2006-06-20        6.0

In [21]:
df[['D1','D2','EXAMDATE','AGE','PTGENDER','PTEDUCAT','PTETHCAT','DX','EXAMDATE_bl','ADAS11_bl']][df['PTID'] =='116_S_0360']

D1  D2    EXAMDATE   AGE PTGENDER  PTEDUCAT         PTETHCAT  DX EXAMDATE_bl  ADAS11_bl
852   0   0  2006-05-16  73.1     Male        18  Not Hisp/Latino  NL  2006-05-16       6.67

In [22]:
df[['D1','D2','EXAMDATE','AGE','PTGENDER','PTEDUCAT','PTETHCAT','DX','EXAMDATE_bl','ADAS11_bl']][df['PTID'] =='126_S_1340']

D1  D2    EXAMDATE   AGE PTGENDER  PTEDUCAT         PTETHCAT   DX EXAMDATE_bl  ADAS11_bl
8904   0   0  2007-04-02  82.4   Female        20  Not Hisp/Latino  MCI  2007-04-02        8.0

### Include only those who have rolled over to the prediction set
Remove also thos who are not in the training set and prediction set
i.e, D1 and D2 = 1

In [23]:
df1 = df[(df['D1'] == 1) & (df['D2'] == 1)]

In [24]:
df1['PTID'].nunique()

882

In [25]:
df1[['D1','D2','EXAMDATE','AGE','PTGENDER','PTEDUCAT','PTETHCAT','DX','EXAMDATE_bl','ADAS11_bl']][df1['PTID'] =='002_S_2073']

D1  D2    EXAMDATE   AGE PTGENDER  PTEDUCAT         PTETHCAT         DX EXAMDATE_bl  ADAS11_bl
3122    1   1  2010-09-21  63.4   Female        20  Not Hisp/Latino        MCI  2010-09-21        6.0
3123    1   1  2010-12-06  63.4   Female        20  Not Hisp/Latino        NaN  2010-09-21        6.0
5524    1   1  2012-09-18  63.4   Female        20  Not Hisp/Latino        MCI  2010-09-21        6.0
5525    1   1  2011-09-29  63.4   Female        20  Not Hisp/Latino        MCI  2010-09-21        6.0
5526    1   1  2011-03-17  63.4   Female        20  Not Hisp/Latino        MCI  2010-09-21        6.0
5691    1   1  2013-10-01  63.4   Female        20  Not Hisp/Latino        MCI  2010-09-21        6.0
9145    1   1  2012-03-26  63.4   Female        20  Not Hisp/Latino        NaN  2010-09-21        6.0
9146    1   1  2013-04-02  63.4   Female        20  Not Hisp/Latino        NaN  2010-09-21        6.0
10934   1   1  2014-04-01  63.4   Female        20  Not Hisp/Latino        NaN  2010-09-21        6.0
11395   1   1  2014-10-07  63.4   Female        20  Not Hisp/Latino  MCI to NL  2010-09-21        6.0
12017   1   1  2015-09-29  63.4   Female        20  Not Hisp/Latino         NL  2010-09-21        6.0
12668   1   1  2016-09-27  63.4   Female        20  Not Hisp/Latino         NL  2010-09-21        6.0

In [26]:
df1.groupby(['PTID','EXAMDATE']).agg(
{ 
 'AGE' : lambda x: list(x)    
})

AGE
PTID       EXAMDATE          
002_S_0295 2006-05-09  [84.8]
           2006-11-02  [84.8]
           2007-05-25  [84.8]
           2007-11-12  [84.8]
           2008-08-04  [84.8]
           2008-11-06  [84.8]
           2009-05-07  [84.8]
           2009-11-18  [84.8]
           2010-05-13  [84.8]
           2010-11-02  [84.8]
           2011-06-02  [84.8]
           2011-11-14  [84.8]
           2012-05-10  [84.8]
002_S_0413 2006-05-08  [76.3]
           2006-11-14  [76.3]
           2007-06-01  [76.3]
           2007-11-28  [76.3]
           2008-07-31  [76.3]
           2008-11-06  [76.3]
           2009-04-30  [76.3]
           2009-11-02  [76.3]
           2010-05-06  [76.3]
           2010-11-02  [76.3]
           2011-06-16  [76.3]
           2011-11-07  [76.3]
           2012-05-15  [76.3]
           2012-11-08  [76.3]
           2013-05-14  [76.3]
           2013-11-20  [76.3]
           2014-05-01  [76.3]
           2015-06-02  [76.3]
002_S_0685 2006-07-20  [89.6]
           2007-01-29  [89.6]
           2007-08-02  [89.6]
           2008-02-03  [89.6]
           2008-08-22  [89.6]
           2009-01-13  [89.6]
           2009-07-23  [89.6]
           2010-01-28  [89.6]
           2010-07-15  [89.6]
           2011-02-02  [89.6]
           2011-07-05  [89.6]
           2012-01-13  [89.6]
           2012-07-27  [89.6]
           2013-01-15  [89.6]
           2013-08-02  [89.6]
           2014-01-28  [89.6]
           2015-07-21  [89.6]
002_S_1155 2007-01-04  [57.8]
           2007-06-21  [57.8]
           2008-01-16  [57.8]
           2008-08-21  [57.8]
           2009-03-13  [57.8]
           2009-07-01  [57.8]
           2010-01-07  [57.8]
           2010-07-20  [57.8]
           2011-01-06  [57.8]
           2011-06-30  [57.8]
           2011-12-23  [57.8]
           2012-07-02  [57.8]
           2012-12-20  [57.8]
           2013-06-27  [57.8]
           2014-01-09  [57.8]
           2015-01-08  [57.8]
           2016-01-12  [57.8]
002_S_1261 2007-02-27  [71.1]
           2007-08-30  [71.1]
           2008-06-09  [71.1]
           2008-09-30  [71.1]
           2009-02-05  [71.1]
           2009-08-26  [71.1]
           2010-02-25  [71.1]
           2010-08-24  [71.1]
           2011-02-24  [71.1]
           2011-08-25  [71.1]
           2012-02-21  [71.1]
           2012-09-07  [71.1]
           2013-03-07  [71.1]
           2014-03-06  [71.1]
           2015-06-02  [71.1]
002_S_1268 2007-03-07  [82.7]
           2007-09-21  [82.7]
           2008-05-12  [82.7]
           2008-09-10  [82.7]
           2009-03-13  [82.7]
           2009-09-09  [82.7]
           2010-03-05  [82.7]
           2011-04-21  [82.7]
           2011-09-09  [82.7]
           2012-03-13  [82.7]
           2012-09-11  [82.7]
           2013-03-14  [82.7]
002_S_1280 2007-02-27  [70.7]
           2007-08-23  [70.7]
           2008-03-25  [70.7]
           2008-09-30  [70.7]
           2009-02-19  [70.7]
           2009-09-01  [70.7]
           2010-02-18  [70.7]
           2010-09-09  [70.7]
...                       ...
941_S_1203 2007-02-22  [83.3]
           2007-08-16  [83.3]
           2008-04-29  [83.3]
           2008-09-30  [83.3]
           2009-02-23  [83.3]
           2009-08-21  [83.3]
           2010-03-19  [83.3]
           2010-09-07  [83.3]
           2011-03-23  [83.3]
           2011-09-09  [83.3]
941_S_2060 2010-10-21  [73.9]
           2010-12-06  [73.9]
           2011-04-20  [73.9]
           2011-10-12  [73.9]
           2012-04-16  [73.9]
           2012-11-20  [73.9]
           2013-05-08  [73.9]
           2013-10-09  [73.9]
           2014-10-23  [73.9]
           2015-10-30  [73.9]
941_S_4036 2011-07-06  [74.0]
           2011-10-06  [74.0]
           2012-01-25  [74.0]
           2012-08-17  [74.0]
           2013-01-17  [74.0]
           2013-08-21  [74.0]
           2014-07-18  [74.0]
           2015-06-26  [74.0]
           2016-06-27  [74.0]
941_S_4100 2011-08-16  [78.5]
           2011-10-03  [78.5]
      

In [27]:
df1.head(2)

RID        PTID VISCODE  SITE  D1  D2 COLPROT ORIGPROT    EXAMDATE DX_bl  DXCHANGE   AGE PTGENDER  PTEDUCAT         PTETHCAT PTRACCAT  PTMARRY  APOE4      FDG  PIB  \
0     2  011_S_0002      bl    11   1   1   ADNI1    ADNI1  2005-09-08    CN       1.0  74.3     Male        16  Not Hisp/Latino    White  Married    0.0  1.36926  NaN   
22    8  011_S_0008      bl    11   1   1   ADNI1    ADNI1  2005-09-19    CN       1.0  84.5   Female        18  Not Hisp/Latino    White  Widowed    0.0  1.27628  NaN   

    AV45  CDRSB  ADAS11  ADAS13  MMSE  RAVLT_immediate  RAVLT_learning  RAVLT_forgetting  RAVLT_perc_forgetting  FAQ  MOCA  EcogPtMem  EcogPtLang  EcogPtVisspat  \
0    NaN    0.0   10.67   18.67  28.0             44.0             4.0               6.0                54.5455  0.0   NaN        NaN         NaN            NaN   
22   NaN    0.0    5.00    7.00  28.0             51.0             7.0               3.0                25.0000  0.0   NaN        NaN         NaN            NaN   

    EcogPtPlan  EcogPtOrgan  EcogPtDivatt  EcogPtTotal  EcogSPMem  EcogSPLang  EcogSPVisspat  EcogSPPlan  EcogSPOrgan  EcogSPDivatt  EcogSPTotal      FLDSTRENG  \
0          NaN          NaN           NaN          NaN        NaN         NaN            NaN         NaN          NaN           NaN          NaN  1.5 Tesla MRI   
22         NaN          NaN           NaN          NaN        NaN         NaN            NaN         NaN          NaN           NaN          NaN  1.5 Tesla MRI   

                                            FSVERSION  Ventricles  Hippocampus  WholeBrain  Entorhinal  Fusiform  MidTemp        ICV  DX EXAMDATE_bl  CDRSB_bl  \
0   Cross-Sectional FreeSurfer (FreeSurfer Version...    118233.0       8336.0   1229740.0      4177.0   16559.0  27936.0  1984660.0  NL  2005-09-08       0.0   
22  Cross-Sectional FreeSurfer (FreeSurfer Version...     18757.0       6080.0    948684.0      4190.0   14043.0  20076.0  1396070.0  NL  2005-09-19       0.0   

    ADAS11_bl  ADAS13_bl  MMSE_bl  RAVLT_immediate_bl  RAVLT_learning_bl  RAVLT_forgetting_bl  RAVLT_perc_forgetting_bl  FAQ_bl   FLDSTRENG_bl  \
0       10.67      18.67       28                44.0                4.0                  6.0                   54.5455     0.0  1.5 Tesla MRI   
22       5.00       7.00       28                51.0                7.0                  3.0                   25.0000     0.0  1.5 Tesla MRI   

                                         FSVERSION_bl  Ventricles_bl  Hippocampus_bl  WholeBrain_bl  Entorhinal_bl  Fusiform_bl  MidTemp_bl     ICV_bl  MOCA_bl  \
0   Cross-Sectional FreeSurfer (FreeSurfer Version...       118233.0          8336.0      1229740.0         4177.0      16559.0     27936.0  1984660.0      NaN   
22  Cross-Sectional FreeSurfer (FreeSurfer Version...        18757.0          6080.0       948684.0         4190.0      14043.0     20076.0  1396070.0      NaN   

    EcogPtMem_bl  EcogPtLang_bl  EcogPtVisspat_bl  EcogPtPlan_bl  EcogPtOrgan_bl  EcogPtDivatt_bl  EcogPtTotal_bl  EcogSPMem_bl  EcogSPLang_bl  EcogSPVisspat_bl  \
0            NaN            NaN               NaN            NaN             NaN              NaN             NaN           NaN            NaN               NaN   
22           NaN            NaN               NaN            NaN             NaN              NaN             NaN           NaN            NaN               NaN   

    EcogSPPlan_bl  EcogSPOrgan_bl  EcogSPDivatt_bl  EcogSPTotal_bl   FDG_bl  PIB_bl  AV45_bl  Years_bl  Month_bl  Month  M           update_stamp  \
0             NaN             NaN              NaN             NaN  1.36926     NaN      NaN       0.0       0.0      0  0  2016-09-02 00:45:03.0   
22            NaN             NaN              NaN             NaN  1.27628     NaN      NaN       0.0       0.0      0  0  2016-09-02 00:45:03.0   

   EXAMDATE_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16 VERSION_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16 FLDSTRENG_UCSFFSL_02_01_16_UCSFFSL51ALL_08_01_16  \


In [28]:
df1['DX'].unique()

array(['NL', 'MCI', 'Dementia', 'NL to MCI', 'MCI to Dementia',
       'MCI to NL', nan, 'NL to Dementia', 'Dementia to MCI'],
      dtype=object)

In [29]:
df1['DX'].value_counts()

MCI                2392
NL                 1994
Dementia            468
MCI to Dementia     163
NL to MCI            79
MCI to NL            62
Dementia to MCI       5
NL to Dementia        1
Name: DX, dtype: int64

In [30]:
len(df1[df1['DX'].isnull()])

2482

In [31]:
df2 = df1.loc[df1['DX'].notnull()]

In [32]:
df2.shape

(5164, 1907)

In [33]:
df2['PTID'].nunique()

882

In [34]:
df2['DX'].unique()

array(['NL', 'MCI', 'Dementia', 'NL to MCI', 'MCI to Dementia',
       'MCI to NL', 'NL to Dementia', 'Dementia to MCI'], dtype=object)

In [35]:
df2 = df2.replace({'NL to MCI': 'MCI', 'MCI to Dementia': 'Dementia', 'MCI to NL' : 'NL', 'NL to Dementia': 'Dementia', 'Dementia to MCI': 'MCI'})

In [36]:
df2['DX'].value_counts()

MCI         2476
NL          2056
Dementia     632
Name: DX, dtype: int64

In [37]:
list(df2.columns)

['RID',
 'PTID',
 'VISCODE',
 'SITE',
 'D1',
 'D2',
 'COLPROT',
 'ORIGPROT',
 'EXAMDATE',
 'DX_bl',
 'DXCHANGE',
 'AGE',
 'PTGENDER',
 'PTEDUCAT',
 'PTETHCAT',
 'PTRACCAT',
 'PTMARRY',
 'APOE4',
 'FDG',
 'PIB',
 'AV45',
 'CDRSB',
 'ADAS11',
 'ADAS13',
 'MMSE',
 'RAVLT_immediate',
 'RAVLT_learning',
 'RAVLT_forgetting',
 'RAVLT_perc_forgetting',
 'FAQ',
 'MOCA',
 'EcogPtMem',
 'EcogPtLang',
 'EcogPtVisspat',
 'EcogPtPlan',
 'EcogPtOrgan',
 'EcogPtDivatt',
 'EcogPtTotal',
 'EcogSPMem',
 'EcogSPLang',
 'EcogSPVisspat',
 'EcogSPPlan',
 'EcogSPOrgan',
 'EcogSPDivatt',
 'EcogSPTotal',
 'FLDSTRENG',
 'FSVERSION',
 'Ventricles',
 'Hippocampus',
 'WholeBrain',
 'Entorhinal',
 'Fusiform',
 'MidTemp',
 'ICV',
 'DX',
 'EXAMDATE_bl',
 'CDRSB_bl',
 'ADAS11_bl',
 'ADAS13_bl',
 'MMSE_bl',
 'RAVLT_immediate_bl',
 'RAVLT_learning_bl',
 'RAVLT_forgetting_bl',
 'RAVLT_perc_forgetting_bl',
 'FAQ_bl',
 'FLDSTRENG_bl',
 'FSVERSION_bl',
 'Ventricles_bl',
 'Hippocampus_bl',
 'WholeBrain_bl',
 'Entorhinal_bl',


In [38]:
demographics = df2[['PTID','EXAMDATE','DX','AGE',
 'PTGENDER',
 'PTEDUCAT',
 'PTETHCAT',
 'PTRACCAT',
 'PTMARRY',
 'APOE4'
]]

In [39]:
#demographics.groupby(['PTID'])['EXAMDATE'].count().sort_values(ascending=False)

In [40]:
demographics['APOE4'].value_counts()

0.0    3136
1.0    1679
2.0     349
Name: APOE4, dtype: int64

In [53]:
demo= demographics.groupby('PTID').agg(
{
 'PTGENDER' : lambda x: max(list(x)),
 'PTEDUCAT': lambda x: max(list(x)),
 'PTETHCAT': lambda x: max(list(x)),
 'PTRACCAT': lambda x: max(list(x)),
 'PTMARRY': lambda x: max(list(x)),
 'APOE4' : lambda x: max(list(x))
    
})

# Using ravel, and a string join, we can create better names for the columns:
demo.columns = ["".join(x) for x in demo.columns.ravel()]

In [54]:
demo = demo.reset_index()

In [58]:
demo.columns

Index(['PTID', 'PTGENDER', 'PTEDUCAT', 'PTETHCAT', 'PTRACCAT', 'PTMARRY', 'APOE4'], dtype='object')

In [59]:
demo[['PTGENDER', 'PTEDUCAT', 'PTETHCAT', 'PTRACCAT', 'PTMARRY', 'APOE4']].apply(pd.Series.value_counts)

PTGENDER  PTEDUCAT  PTETHCAT  PTRACCAT  PTMARRY  APOE4
0.0                     NaN       NaN       NaN       NaN      NaN  522.0
1.0                     NaN       NaN       NaN       NaN      NaN  298.0
2.0                     NaN       NaN       NaN       NaN      NaN   62.0
6                       NaN       1.0       NaN       NaN      NaN    NaN
8                       NaN       5.0       NaN       NaN      NaN    NaN
10                      NaN       4.0       NaN       NaN      NaN    NaN
11                      NaN       4.0       NaN       NaN      NaN    NaN
12                      NaN      95.0       NaN       NaN      NaN    NaN
13                      NaN      35.0       NaN       NaN      NaN    NaN
14                      NaN      95.0       NaN       NaN      NaN    NaN
15                      NaN      45.0       NaN       NaN      NaN    NaN
16                      NaN     204.0       NaN       NaN      NaN    NaN
17                      NaN      49.0       NaN       NaN      NaN    NaN
18                      NaN     171.0       NaN       NaN      NaN    NaN
19                      NaN      46.0       NaN       NaN      NaN    NaN
20                      NaN     128.0       NaN       NaN      NaN    NaN
Am Indian/Alaskan       NaN       NaN       NaN       2.0      NaN    NaN
Asian                   NaN       NaN       NaN      12.0      NaN    NaN
Black                   NaN       NaN       NaN      35.0      NaN    NaN
Divorced                NaN       NaN       NaN       NaN     92.0    NaN
Female                408.0       NaN       NaN       NaN      NaN    NaN
Hawaiian/Other PI       NaN       NaN       NaN       1.0      NaN    NaN
Hisp/Latino             NaN       NaN      32.0       NaN      NaN    NaN
Male                  474.0       NaN       NaN       NaN      NaN    NaN
Married                 NaN       NaN       NaN       NaN    653.0    NaN
More than one           NaN       NaN       NaN      11.0      NaN    NaN
Never married           NaN       NaN       NaN       NaN     33.0    NaN
Not Hisp/Latino         NaN       NaN     845.0       NaN      NaN    NaN
Unknown                 NaN       NaN       5.0       2.0      5.0    NaN
White                   NaN       NaN       NaN     819.0      NaN    NaN
Widowed                 NaN       NaN       NaN       NaN     99.0    NaN

#### The demographics of the patients

In [60]:
df3 = demographics[['PTID','EXAMDATE','DX','AGE']]

In [64]:
df3['EXAMDATE'] = pd.to_datetime(df3['EXAMDATE'])

In [85]:
dfTest

PTID   EXAMDATE  DX   AGE
53    011_S_0023 2005-11-08  NL  71.7
54    011_S_0023 2006-05-08  NL  71.7
55    011_S_0023 2006-11-21  NL  71.7
56    011_S_0023 2007-11-20  NL  71.7
57    011_S_0023 2008-11-25  NL  71.7
58    011_S_0023 2011-12-20  NL  71.7
59    011_S_0023 2012-12-27  NL  71.7
4958  011_S_0023 2010-12-21  NL  71.7
4965  011_S_0023 2010-01-06  NL  71.7
5776  011_S_0023 2013-12-18  NL  71.7

In [81]:
dfTest = df3[['PTID','EXAMDATE','DX','AGE']][df3['PTID'] == '011_S_0023']

In [86]:
dfTest['AGE'].iloc[1] = dfTest['AGE'].iloc[1] +  ((dfTest['EXAMDATE'].iloc[1] - (dfTest['EXAMDATE'].iloc[0])).days / 365)

In [87]:
dfTest

PTID   EXAMDATE  DX       AGE
53    011_S_0023 2005-11-08  NL  71.70000
54    011_S_0023 2006-05-08  NL  72.19589
55    011_S_0023 2006-11-21  NL  71.70000
56    011_S_0023 2007-11-20  NL  71.70000
57    011_S_0023 2008-11-25  NL  71.70000
58    011_S_0023 2011-12-20  NL  71.70000
59    011_S_0023 2012-12-27  NL  71.70000
4958  011_S_0023 2010-12-21  NL  71.70000
4965  011_S_0023 2010-01-06  NL  71.70000
5776  011_S_0023 2013-12-18  NL  71.70000

In [113]:
df3['Difference'] = (np.where(df3.PTID == df3.PTID.shift(),((df3['EXAMDATE'] - df3['EXAMDATE'].shift()).dt.days) / 365, 0))

In [121]:
df3['AGE_new'] = np.where(df3.PTID == df3.PTID.shift(),(df3['AGE_new']) + df3['Difference'].shift(), df3['AGE'])

In [124]:
 df3[['PTID','EXAMDATE','DX','AGE_new']][df3['PTID'] == '011_S_0023']

PTID   EXAMDATE  DX    AGE_new
53    011_S_0023 2005-11-08  NL  71.700000
54    011_S_0023 2006-05-08  NL  72.195890
55    011_S_0023 2006-11-21  NL  73.231507
56    011_S_0023 2007-11-20  NL  73.776712
57    011_S_0023 2008-11-25  NL  74.710959
58    011_S_0023 2011-12-20  NL  76.801370
59    011_S_0023 2012-12-27  NL  78.858904
4958  011_S_0023 2010-12-21  NL  71.700000
4965  011_S_0023 2010-01-06  NL  71.700000
5776  011_S_0023 2013-12-18  NL  71.700000

#### Sort the dates

#### Modelling

We used national Veterans Health Administration (VHA) data to identify CHC patients in care between 2000-2016. The primary outcome was cirrhosis development ascertained by two consecutive aspartate aminotransferase (AST)-to-platelet ratio indexes (APRIs) > 2 after time zero given the infrequency of liver biopsy in clinical practice and that APRI is a validated non-invasive biomarker of fibrosis in CHC. We excluded those with initial APRI > 2 or pre-existing diagnosis of cirrhosis, hepatocellular carcinoma or hepatic decompensation. Enrollment was defined as the date of the first APRI. Time zero was defined as 2 years after enrollment. Cross-sectional (CS) models used predictors at or closest before time zero as a comparison. Longitudinal models used CS predictors plus longitudinal summary variables (maximum, minimum, maximum of slope, minimum of slope and total variation) between enrollment and time zero. Covariates included demographics, labs, and body mass index. Model performance was evaluated using concordance and area under the receiver operating curve (AuROC). A total of 72,683 individuals with CHC were analyzed with the cohort having a mean age of 52.8, 96.8% male and 53% white. There are 11,616 individuals (16%) who met the primary outcome over a mean follow-up of 7 years. We found superior predictive performance for the longitudinal Cox model compared to the CS Cox model (concordance 0.764 vs 0.746), and for the longitudinal boosted-survival-tree model compared to the linear Cox model (concordance 0.774 vs 0.764). The accuracy of the longitudinal models at 1,3,5 years after time zero also showed superior performance compared to the CS model, based on AuROC.

https://www.ncbi.nlm.nih.gov/pubmed/30608929

https://www.biorxiv.org/content/biorxiv/early/2018/06/04/338913.full.pdf ( with the code)